src/main/java/org/owasp/esapi/reference/DefaultValidator.java

```diff
@@ -119,42 +119,54 @@ public DefaultValidator( Encoder encoder ) {
 
 
     /**
-     * Add a validation rule to the registry using the "type name" of the rule as the key.
+     * {@inheritDoc}
      */
-    public void addRule( ValidationRule rule ) {
+    @Override
+    public void addRule(ValidationRule rule ) {
         rules.put( rule.getTypeName(), rule );
     }
 
     /**
-     * Get a validation rule from the registry with the "type name" of the rule as the key.
+     * {@inheritDoc}
      */
-    public ValidationRule getRule( String name ) {
+    @Override
+    public ValidationRule getRule(String name ) {
         return rules.get( name );
     }
 
 
     /**
-     * Returns true if data received from browser is valid. Double encoding is treated as an attack. The
-     * default encoder supports html encoding, URL encoding, and javascript escaping. Input is canonicalized
-     * by default before validation.
-     *
-     * @param context A descriptive name for the field to validate. This is used for error facing validation messages and element identification.
-     * @param input The actual user input data to validate.
-     * @param type The regular expression name while maps to the actual regular expression from "ESAPI.properties".
-     * @param maxLength The maximum post-canonicalized String length allowed.
-     * @param allowNull If allowNull is true then a input that is NULL or an empty string will be legal. If allowNull is false then NULL or an empty String will throw a ValidationException.
-     * @return The canonicalized user input.
-     * @throws IntrusionException
+     * {@inheritDoc}
+     * <p>
+     * Double encoding is treated as an attack.
+     * The default encoder supports HTML encoding, URL encoding, and javascript escaping.
```

Comment:
One important aspect that is not mentioned anywhere here is that the behavior of much of this entire implementation class is driven by these 3 tunable ESAPI properties in the **ESAPI.properties** file. Those properties and their default values are as follows:

```
Encoder.AllowMultipleEncoding=false
Encoder.AllowMixedEncoding=false
Encoder.DefaultCodecList=HTMLEntityCodec,PercentCodec,JavaScriptCodec
```

I think this should be documented / explained somewhere. If I were writing it, I would mention this at the 'class' javadoc level, but I will leave it up to you where you think it is most suitable.

Reply:
I'm still thinking of suitable words

Reply:
No worries. I'll merge this and then add something in a separate PR.